subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c

```diff
@@ -0,0 +1,1654 @@
+	}
+
+	return NULL;
+}
+
+static int json_atribute_decode(struct lwm2m_input_context *in, struct json_in_formatter_data *fd)
+{
+	uint8_t attrbute_name[3];
+
+	if (fd->name_len == 0 || fd->name_len > 3) {
+		if (fd->name_len == 0 && in->block_ctx && (fd->json_flags & T_VALUE) &&
+		    (fd->json_flags & T_STRING_END)) {
+			return SENML_JSON_STRING_BLOCK_DATA;
+		}
+		return SENML_JSON_UNKNOWN_ATTRIBUTE;
+	}
+
+	if (buf_read(attrbute_name, fd->name_len, CPKT_BUF_READ(in->in_cpkt), &fd->name_offset) <
+	    0) {
+		LOG_ERR("Error parsing attribute name!");
+		return SENML_JSON_UNKNOWN_ATTRIBUTE;
+	}
+
+	if (fd->name_len == 1) {
+		if (attrbute_name[0] == 'n') {
+			return SENML_JSON_NAME_ATTRIBUTE;
+		} else if (attrbute_name[0] == 't') {
+			return SENML_JSON_TIME_ATTRIBUTE;
+		} else if (attrbute_name[0] == 'v') {
+			return SENML_JSON_FLOAT_VALUE_ATTRIBUTE;
+		}
+
+	} else if (fd->name_len == 2) {
+		if (attrbute_name[0] == 'b') {
+			if (attrbute_name[1] == 'n') {
+				return SENML_JSON_BASE_NAME_ATTRIBUTE;
+			} else if (attrbute_name[1] == 't') {
+				return SENML_JSON_BASE_TIME_ATTRIBUTE;
+			}
+		} else if (attrbute_name[0] == 'v') {
+			if (attrbute_name[1] == 'b') {
+				return SENML_JSON_BOOLEAN_VALUE_ATTRIBUTE;
+			} else if (attrbute_name[1] == 'd') {
+				return SENML_JSON_OPAQUE_VALUE_ATTRIBUTE;
+			} else if (attrbute_name[1] == 's') {
+				return SENML_JSON_STRING_VALUE_ATTRIBUTE;
+			}
+		}
+	} else if (fd->name_len == 3) {
+		if (attrbute_name[0] == 'v' && attrbute_name[1] == 'l' && attrbute_name[2] == 'o') {
+			return SENML_JSON_OBJ_LINK_VALUE_ATTRIBUTE;
+		}
+	}
+
+	return SENML_JSON_UNKNOWN_ATTRIBUTE;
+}
+
+/* Parse SenML attribute & value pairs  */
+static int json_next_token(struct lwm2m_input_context *in, struct json_in_formatter_data *fd)
+{
+	uint8_t cont, c = 0;
+	bool escape = false;
+	struct lwm2m_senml_json_context *block_ctx;
+
+	(void)memset(fd, 0, sizeof(*fd));
+	cont = 1U;
+	block_ctx = seml_json_context_get(in->block_ctx);
+
+	if (block_ctx && block_ctx->json_flags) {
+		/* Store from last sequence */
+		fd->json_flags = block_ctx->json_flags;
+		block_ctx->json_flags = 0;
+	}
+
+	/* We will be either at start, or at a specific position */
+	while (in->offset < in->in_cpkt->offset && cont) {
+		fd->offset = in->offset;
+		if (buf_read_u8(&c, CPKT_BUF_READ(in->in_cpkt), &in->offset) < 0) {
+			break;
+		}
+
+		if (c == '\\') {
+			escape = true;
+			/* Keep track of the escape codes */
+			json_add_char(in, fd);
+			continue;
+		}
+
+		switch (c) {
+		case '[':
+			if (!escape) {
+				fd->json_flags |= T_OBJECT_BEGIN;
+				cont = 0U;
+			} else {
+				json_add_char(in, fd);
+			}
+			break;
+		case '}':
+		case ']':
+			if (!escape) {
+				fd->json_flags |= T_OBJECT_END;
+				cont = 0U;
+			} else {
+				json_add_char(in, fd);
+			}
+			break;
+		case '{':
+			if (!escape) {
+				fd->json_flags |= T_OBJECT_BEGIN;
+			} else {
+				json_add_char(in, fd);
+			}
+
+			break;
+
+
```

Comment:
Why there are so big gaps in the switch-case

Reply:
I will fix this.